processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -158,12 +160,33 @@ public GroupByQueryQueryToolChest(
       }
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
-      final List<AggregatorFactory> aggs = Lists.newArrayList();
-      for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
+
+      // check that all fieldName parameters in the outer query match up with a name parameter in the inner query
+      // for an aggregator or a post aggregator
+      Set<String> innerFieldNames = Sets.newHashSet();
+      for (AggregatorFactory innerAggregator : subquery.getAggregatorSpecs()) {
+        innerFieldNames.add(innerAggregator.getName());
+      }
+      for (PostAggregator innerPostAggregator : subquery.getPostAggregatorSpecs()) {
+        innerFieldNames.add(innerPostAggregator.getName());
+      }
+
+      for (AggregatorFactory outerAggregator : query.getAggregatorSpecs()) {
+        for (final String fieldName : outerAggregator.requiredFields()) {
+          if (!innerFieldNames.contains(fieldName)) {
+            throw new IllegalArgumentException(
+                String.format("Subquery must have an aggregator or post aggregator with name '%s'", fieldName)
+            );
+          }
+        }
       }
 
       // We need the inner incremental index to have all the columns required by the outer query
+      final List<AggregatorFactory> aggs = Lists.newArrayList(subquery.getAggregatorSpecs());
+      for (PostAggregator postAgg : subquery.getPostAggregatorSpecs()) {
+        aggs.add(new CountAggregatorFactory(postAgg.getName())); // aggregator type doesn't matter here
```

Comment:
As discussed, this just causes the results of the post aggregator from the subquery to be materialized in the incremental index so that it can be queried again by the outer query.